public/js/medid/creator.js

```diff
@@ -0,0 +1,106 @@
+define(['jquery'], function($) {
+
+  var creator = {
+    table: $('#fields tbody'),
+    saveEndpoint: "",
+    getMethod: function (getFnc) {
+      this.getPDF = getFnc;
+    },
+    downloadMethod: function (downloadFnc) {
+      this.downloadPDF = downloadFnc;
+    }
+  }
+
+  // Retrieval method for the field data
+	creator.fields = function () {
+		var fields = [];
+
+		var label, field;
+		creator.table.children('tr').each(function () {
+			label = $(this).find('.medid-label').val();
+			field = $(this).find('.medid-field').val();
+
+			//if (label == "") {
+      //  fields.push({field: field});
+			//} else {
+			//	fields.push({label: label, field: field})
+			//}
+
+      fields.push({label: label, field: field});
+
+      // Thijs: We might want to get rid of this part
+      if (label == "Name") {
+        creator.userName = field;
+      }
+		});
+    return fields;
+	}
+
+  creator.addField = function (label, field) {
+    var field = $("<tr><td><input class='medid-label' maxlength='15' value='" + label + "' type='text' /></td><td><input class='medid-field' maxlength='200' type='text' value='" + field + "' /></td><td><input class='removeField' type='button' value='Remove' /></td></tr>");
```

Comment:
Have a look into Mustache.js, it’s a template system it will give more flexibility in this case. You already use handlebars which is based on Mustache.